spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -286,6 +327,38 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
 		return adapter;
 	}
 
+	private MessageProducer createKclConsumerEndpoint(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<KinesisConsumerProperties> properties) {
+		KinesisConsumerProperties kinesisConsumerProperties = properties.getExtension();
+
+		String shardIteratorType = kinesisConsumerProperties.getShardIteratorType();
+
+		Region parsedRegion = (this.configurationProperties.getRegion() == null) ?
+				null : Region.of(this.configurationProperties.getRegion());
+		KclMessageDrivenChannelAdapter adapter = new KclMessageDrivenChannelAdapter(
+				destination.getName(), Executors.newSingleThreadExecutor(), parsedRegion);
```

Comment:
The same concern about `region` here.
Also this `executor` should be as a `@Bean` as well, conditional on the `kplKclEnabled` probably.
Although the same `exectutor` could be injected into the `KinesisMessageDrivenChannelAdapter` as well...
Also would be better to reuse `this.amazonKinesis` instead of an internal instance. That way we would inherit credentials and region from the auto-configuration.
Possibly we can expose beans for the `CloudWatchAsyncClient` as well.
We have already a bean for the `DynamoDbAsyncClient` though...

Reply:
We can't use the existing kinesis and dynamoDB clients since KCL uses the amazon SDK V2 and they are not compatible.
I can add beans for the new version of the kinesis and dynamo clients, conditioned on the `kplKclEnabled` property too, and initialize them with the region from the auto-configuration.
However, I can't use the credentials provider from the auto-configuration and the new version of it is not currently available.
There's an issue opened for this in spring-cloud-aws, but I don't know if someone is working on it : https://github.com/spring-cloud/spring-cloud-aws/issues/267

You want me to add the beans without the credientials provider configuration for now ?

Reply:
Oh!  AWS SDK V2! 😮 

Well, then we can't have this feature until we align all the API with that new version.
I see that KPL is still on SDK V1 anyway. Any news when that is going to be updated respectively?
Can't we just downgrade SI-AWS `KclMessageDrivenChannelAdapter` to be based on the SDK V1 for time being?

Thanks for understanding!

Reply:
Well, since the V1 and V2 of the SDK use different base packages, they could both be used in the same project, but if you prefer not to mix the versions, we can downgrade KCL to v1.9.3.
I don't know when KPL is going to support the V2, I didn't see any open issue for that.

If we downgrade, I'd like to wait until you have time to review the changes for the checkpoint modes if you don't mind to avoid the merge.

Reply:
I understand what you mean with the `different base packages`, but this is going to be a burden for end-user to select a proper SDK from classpath when he/she develops a target project.
Would be better to have everything aligned.

When Spring Cloud AWS moves to SDK V2, I have an idea about Reactive Streams wrappers around that API, so our components will get a gain to deal with `Flux` and `Mono` in non-blocking manner.
More over as you see with support both versions in the project we get a pain to increase an amount of auto-configured beans we would like to expose.
Also it looks like we would need to extract some `CredentialsProvider` adapter to be able to remap auto-configured one to this SDK V2.
Therefore my point to stay with SDK V1 for time being until everything upstream is ready and clean.

More over it is out of our policy to introduce such a major (SDK V2) change in the minor release.

Does it all make sense to you?

Reply:
Ok, I'll check with the team when we can find time to do and test the downgrade, I'll let you know.

Reply:
We just had a meeting to check our planning. We are still willing to do this, but we probably won't be able to work on it for the next 3 weeks.
Is there already an estimated release date for SI-AWS V2.1.0 ?

Reply:
That's OK with us and no, we don't have plans for SI-AWS V2.1.0.
I would like to have over there AWS SDK V2, but that might be too ambitious and we should deffer that until the next version, maybe even next year, when all the upstream API are based on V2 already...